src/citrine/resources/ara_definition.py

```diff
@@ -124,6 +126,74 @@ def add_columns(self, *,
             columns=copy(self.columns) + columns
         )
 
+    def add_all_ingredients(self, *,
+                            process_template: LinkByUID,
+                            project,
+                            quantity_dimension: IngredientQuantityDimension,
+                            scope: str = 'id'
+                            ):
+        """[ALPHA] Add variables and columns for all of the possible ingredients in a process.
+
+        For each allowed ingredient name in the process template there is a column for the if of
+        the ingredient and a column for the quantity of the ingredient. If the quantities are
+        given in absolute amounts then there is also a column for units.
+
+        Parameters
+        ------------
+        process_template: LinkByUID
+            scope and id of a registered process template
+        project: Project
+            a project that has access to the process template
+        quantity_dimension: IngredientQuantityDimension
+            the dimension in which to report ingredient quantities
+        scope: Optional[str]
+            the scope for which to get ingredient ids (default is Citrine scope, 'id')
+
+        """
+        quantity_dimension_display = {
+            IngredientQuantityDimension.ABSOLUTE: "absolute quantity",
+            IngredientQuantityDimension.MASS: "mass fraction",
+            IngredientQuantityDimension.VOLUME: "volume fraction",
+            IngredientQuantityDimension.NUMBER: "number fraction"
+        }
+        process: ProcessTemplate = project.process_templates.get(
+            uid=process_template.id, scope=process_template.scope)
+        new_variables = []
+        new_columns = []
+        for name in process.allowed_names:
+            identifier_variable = IngredientIdentifierByProcessTemplateAndName(
+                name="{} of ingredient {} in process {}".format(scope, name, process.name),
+                headers=[process.name, name + ' ' + scope],
+                process_template=process_template,
+                ingredient_name=name,
+                scope=scope
+            )
+            quantity_variable = IngredientQuantityByProcessAndName(
+                name="{} of ingredient {} in process {}".format(
+                    quantity_dimension_display[quantity_dimension], name, process.name),
+                headers=[process.name, name, quantity_dimension_display[quantity_dimension]],
```

Comment:
These and the headers above are editorial decisions. Maybe we should let the user provide their own headers? On the other hand they are using a convenience method, so reducing the number of decisions they have to make seems like it could be justified.

Reply:
If we're only using the convenience method route, then that sounds like an unnecessary option, at least in MVP.